.github/workflows/build_and_test.yml

```diff
@@ -101,6 +113,31 @@ jobs:
           flags: unittests
           fail_ci_if_error: false # optional (default = false)
 
+      - name: 'Publish a snapshot to GitHub'
+        id: publish-github
+        if: ${{ github.event_name == 'push' && github.ref_type == 'branch' && github.ref == 'refs/heads/main' && matrix.os == 'macos-latest' }}
```

Comment:
why `macos-latest`?

Reply:
as I remember `macos-latest` builds all required native targets.
but I'm still experimenting on it